### DISCUSSION TITLE
Tell the user when a new fwupd is required

### DISCUSSION THREAD

**Discussion by hughsie** [fwupd/fwupd#6793] [Ideas]
An idea someone suggested yesterday. At the moment, on refresh we have:

    Successfully downloaded new metadata: 5 local devices supported

We might be in the situation where we're missing security updates because the local fwupd version is too old. What about something like:

    Successfully downloaded new metadata: 5 local devices supported, 3 more would be supported with fwupd >= 1.9.13

Alternatively, we could do something like:

	├─UEFI Device Firmware:
	│     Device ID:          f37fb01122dd62c773f4e84ec89737e059712d59
	│     Summary:            UEFI ESRT device
	│     Current version:    24580
	│     Minimum Version:    1
	│     Vendor:             DMI:LENOVO
	│     Problems:           • Requires a newer fwupd
	│     GUID:               33967546-da89-4c51-9c95-5242bcb854e8
	│     Device Flags:       • Internal device
	│     Device Requests:    • Message

Other ideas welcome!

**Comment by superm1** [comment]
Various brain dump of thoughts:

1. To me it feels like a bandaid for distros that don't update frequently enough (looking at Ubuntu LTS and Debian stable).

2. To an average user who sees such a message how do we really want them to solve it? I feel like it's going to mean pushing the snap/flatpak harder. 

Or maybe we just spin a standalone AppImage for them to use in this situation?

3. It isn't going to help for the situation that you don't have a plugin in the first place. How often are things really gated on a version instead of baseline functionality?

**Comment by hughsie** [comment]
> To me it feels like a bandaid for distros that don't update frequently enough

Do you think enough _user pressure_ would affect the policy of LTS/stable distros?

> how do we really want them to solve it

That's the bit I'm also struggling with; there's no user action that means the user can fix the problem. That said, at the moment people have no ideas that they're missing security updates because the client is too old. Maybe it's something we shout return in get-updates? e.g. 

	LENOVO 20TJS2F44A
	│
	└─System Firmware:
	  │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
	  │ 
	  └─ThinkPad P1 Gen 3/X1 Extreme 3rd System Update:
		New version:      0.1.31
		Release Flags:    • Trusted metadata
				  • Is upgrade
				  • Requires a new fwupd version to deploy
		Issue:            CVE-2023-5058

> It isn't going to help for the situation that you don't have a plugin in the first place

Agree, but I don't think we can do anything in that case as we don't enumerate the device. I think in the case of dbx updates the fwupd version is quite a common reason why it can't be deployed.

**Reply by superm1** [reply]
The biggest concern in stable updates is always test coverage.  Look at the last few 1.9.x releases that we've put out and the breakage that wasn't identified until more people used them and identified the issues.  If you extrapolate that out to something that `${LARGE_NUMBER}` of people use in a stable release that only does updates say once a year it's pretty unfortunate if you get stuck with such a release as your stable update.

Put your RHEL maintainer hat on for a minute and think about this hypothetical (admittedly the versions are made up).
Say RHEL 9.3 has fwupd 1.7.9 and you locked down RHEL 9.5  with fwupd 1.9.12.  What would it look like if someone went from RHEL 9.3 to RHEL 9.5?  They'll hit the config bug, the daemon won't start.  This case is only caught by people who upgraded, a fresh install would never hit it.

So I really think the right argument that a stable release is stable is it's stable because it's got enough testing coverage.  I think it's great we have a config migration test now so at least stuff in `/etc` from an upgrade is no longer a blind spot for us.  But I think the next area we have to tackle is wider device emulation to be able to exercise more daemon and plugin flows.

> Do you think enough user pressure would affect the policy of LTS/stable distros?

I think we should approach them separately.

At least in the case of Ubuntu the generic SRU policy is described [here](https://wiki.ubuntu.com/StableReleaseUpdates).
Of most important note is the exception process:
```
The [Technical Board resolution on Landscape](https://lists.ubuntu.com/archives/ubuntu-devel-announce/2009-March/000550.html) provides a general rationale for the types of special cases that may be approved here in future. Most exception approvals are now handled directly by the SRU team.

To obtain a new ongoing exception such as those documented below:

Draft a wiki page, like the ones below, outlining what you believe should be the exception.
Submit it to the SRU team for approval. This can be done to any individual member of the SRU team directly, or you can send it to [ubuntu-release@lists.ubuntu.com](mailto:ubuntu-release@lists.ubuntu.com) for review.
```

And we actually already **have** an [exception for fwupd](https://wiki.ubuntu.com/StableReleaseUpdates#fwupd_and_fwupdate), but it's stale to what has happened the past few years.

So for Ubuntu I think what we should probably do is make a copy of the [exception wiki page](https://wiki.ubuntu.com/firmware-updates) and then update the text to what we _want_ and then request the SRU team to update ratify it.  If they do things should be at least able to go more smoothly.  You still need someone to actually package the SRU backports and upload it.  If we can get a new policy approved maybe this is something that @KaiChuan-Hsieh can help to champion.

> Agree, but I don't think we can do anything in that case as we don't enumerate the device. I think in the case of dbx updates the fwupd version is quite a common reason why it can't be deployed.

TBH I think this is a design problem if the uefi-dbx plugin needs to be updated when there is a new DBX file.

**Reply by hughsie** [reply]
> and then update the text to what we want and then request the SRU team to update ratify it

Right, agree. Is this in your purview?

> if the uefi-dbx plugin needs to be updated when there is a new DBX file

Ahh, not for each release, but when we decide to check something new. This is more of the case where we need 1.9.x and we're on 1.7.x.

**Reply by superm1** [reply]
Once we have some text I can contact the release team for sure.

What is our actual ask? Latest lts always gets stable version? Or something tied to EOL dates? Should 1.7.x be EOL now or soon?

The part that I struggle with is how to justify the "risk of regressions" justifies updating.

**Reply by superm1** [reply]
Here's my stab at a new page.  It's a wiki; please feel free to make changes to it.
https://wiki.ubuntu.com/firmware-updates-2.0

I think to support it we should make a few changes to SECURITY.md.
1. We should post what we expect the EOL date for 1.7.x and 1.8.x
2. The check marks for main, 1.9.x and 1.8.x/1.7.x are different but there is no legend.

**Reply by hughsie** [reply]
Is https://github.com/fwupd/fwupd/pull/6794 insane?

**Reply by superm1** [reply]
Can you review the wiki page? I want to make sure we're totally on the same page before I try to go make a case for a blanket policy exception.

**Reply by hughsie** [reply]
Ohh I did; it makes sense to me!
